workers/unity/Packages/io.improbable.gdk.core/Worker/ConnectionHandlers/SpatialOSConnectionHandler/SpatialOSConnectionHandler.cs

```diff
@@ -36,12 +36,14 @@ public List<string> GetWorkerAttributes()
 
         public void GetMessagesReceived(ref ViewDiff viewDiff)
         {
+            commandMetaData.FlushRemovedIds();
```

Comment:
Does this only get called once a frame or can this have bad side effects if it's called before everything is handled? Is there any way to guard or error against this? Or could that be intended behaviour?

Reply:
This is just cleaning up garbage data

Reply:
When does it become garbage data? I'm wondering if re-entering into this function and flushing it before it becomes garbage is a possibility.

Reply:
`CommandMetaData` is a storage provider for linking a send Request (data) to the request id we got back when the request was send over the network.
Whenever the response comes back, we use the request id that comes with it to lookup the initial Request through `CommandMetaData` to include in the command `Response` structure for further processing. At this point the request ID is marked for removal as it has been processed.

This could likely be refactored to simply remove the request id straight away, instead of a mark and delete.

Reply:
Updated the PR delete immediately